### PR TITLE
r.kappa: fix kappa variance calculation for edge cases

### PR DESCRIPTION
**Pull request by marisn**
The issue was not revealed by tests as tests passed bogous `precision` value to `keyvalue_equals`.

Discovered by an accident as a part of review of https://github.com/OSGeo/grass/pull/6985

### CODE REVIEW

**Review by echoix — DISMISSED** [review]

File: raster/r.kappa/testsuite/test_r_kappa.py [hunk 1]
```diff
@@ -301,6 +301,8 @@ def test_standard_output(self):
 class JSONOutputTest(TestCase):
     """Test printing of parameters in JSON format"""
 
+    precision = 0.001
```
Comment (echoix) [inline raster/r.kappa/testsuite/test_r_kappa.py]: Do you want a smaller value than 1e-3, like in your comment on the other pr?

Comment (marisn) [reply raster/r.kappa/testsuite/test_r_kappa.py]: This will have to be bumped up as soon as  #6977 or #6978 is merged, as current code has `%.5f` for JSON output and thus doesn't matter that much now.

Comment (marisn) [reply raster/r.kappa/testsuite/test_r_kappa.py]: Here you go.

**Comment by echoix** [comment]
Here you go
